Additional Land perfmetrics

bettina-gier opened the pull request:
Before you start, read [CONTRIBUTING.md](https://github.com/ESMValGroup/ESMValTool/blob/master/CONTRIBUTING.md) and the [guide for diagnostic developers](https://esmvaltool.readthedocs.io/en/latest/esmvaldiag/index.html).

Please discuss your idea with the development team before getting started, to avoid disappointment later. The way to do this is to open a new issue on GitHub. If you are planning to modify an existing functionality, please discuss it with the original author(s) by tagging them in the issue.

* * *

**Tasks**

-   [ ] [Create an issue](https://github.com/ESMValGroup/ESMValTool/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [x] Give this pull request a descriptive title that can be used as a one line summary in a changelog
-   [x] Make sure your code is composed of functions of no more than 50 lines and uses meaningful names for variables
-   [x] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [x] Preferably Codacy code quality checks pass, however a few remaining hard to solve Codacy issues are still acceptable. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.
-   [x] Please use `yamllint` to check that your YAML files do not contain mistakes 
-   [x] Add documentation for the recipe to the `doc/sphinx/source/recipes` folder and add a new entry to index.rst

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *
Adding recipe_perfmetrics_land_CMIP5.yml, used for ESMValTool paper Part 2! I realize it's just using other variables than the regular perfmetrics recipe, but as it's used for the paper I assume it'd be nice to have it in the master for anyone wanting to start from any of the paper figures. Don't think additional documentation is required, as I don't see any for the CDS perfmetrics recipe.

bouweandela commented:
> Don't think additional documentation is required

It would be good if you could mention the recipe in the list of recipes here: https://esmvaltool.readthedocs.io/en/latest/recipes/recipe_perfmetrics.html#available-recipes-and-diagnostics, so at least people can find it without looking at the esmvaltool source code.

bettina-gier commented:
Had forgotten about this till now, sorry! Added land_perf to the perfmetrics documentation, also fixed the reformat scripts links in there as they were pointing to the old folder - I assume this might still be wrong in some other recipe documentations? Didn't check all.
Also realize "carbon_perfmetrics" might be a better name due to the inclusion of fgco2, but keeping this as it's what is in the paper with this name and these variables.

mattiarighi commented on esmvaltool/recipes/recipe_perfmetrics_land_CMIP5.yml:
```diff
@@ -0,0 +1,661 @@
+    Recipe for plotting the performance metrics for the CMIP5 datasets,
+    including only land variables
+
+  authors:
```
```suggestion
  authors:
    - gier_bettina
```

mattiarighi commented on esmvaltool/recipes/recipe_perfmetrics_land_CMIP5.yml:
```diff
@@ -0,0 +1,661 @@
+    - cmug
+
+preprocessors:
+
```
```suggestion
```
